src/utils/riff-raff-yaml-file.test.ts

```diff
@@ -0,0 +1,323 @@
+          contentDirectory: my-lambda-artifact
+          parameters:
+            bucketSsmLookup: true
+            lookupByTags: true
+            fileName: my-lambda-artifact.zip
+          actions:
+            - uploadLambda
+        my-application-stack-cfn-test-eu-west-1:
+          type: cloud-formation
+          regions:
+            - eu-west-1
+          stacks:
+            - test
+          app: my-application-stack
+          contentDirectory: /tmp/cdk.out
+          parameters:
+            templateStagePaths:
+              TEST: test-stack.template.json
+          dependencies:
+            - my-lambda-upload-test-eu-west-1
+        my-lambda-update-test-eu-west-1:
+          type: aws-lambda
+          stacks:
+            - test
+          regions:
+            - eu-west-1
+          app: my-lambda
+          contentDirectory: my-lambda-artifact
+          parameters:
+            bucketSsmLookup: true
+            lookupByTags: true
+            fileName: my-lambda-artifact.zip
+          actions:
+            - updateLambda
+          dependencies:
+            - my-application-stack-cfn-test-eu-west-1
+      "
+    `);
+  });
+
+  it("Should add autoscaling deployments", () => {
+    const app = new App({ outdir: "/tmp/cdk.out" });
+
+    class MyApplicationStack extends GuStack {
+      // eslint-disable-next-line custom-rules/valid-constructors -- unit testing
+      constructor(app: App, id: string, props: GuStackProps) {
+        super(app, id, props);
+
+        const appName = "my-app";
+
+        new GuEc2App(this, {
+          app: appName,
+          instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MICRO),
+          access: { scope: AccessScope.PUBLIC },
+          userData: {
+            distributable: {
+              fileName: `${appName}.deb`,
+              executionStatement: `dpkg -i /${appName}/${appName}.deb`,
+            },
+          },
+          certificateProps: {
+            domainName: "rip.gu.com",
+          },
+          monitoringConfiguration: { noMonitoring: true },
+          scaling: {
+            minimumInstances: 1,
+          },
+          applicationPort: 9000,
+          imageRecipe: "arm64-bionic-java11-deploy-infrastructure",
+        });
+      }
+    }
+
+    new MyApplicationStack(app, "test-stack", { stack: "test", stage: "TEST", env: { region: "eu-west-1" } });
+
+    const actual = new RiffRaffYamlFile(app).toYAML();
+
+    expect(actual).toMatchInlineSnapshot(`
+      "allowedStages:
+        - TEST
+      deployments:
+        my-application-stack-cfn-test-eu-west-1:
+          type: cloud-formation
+          regions:
+            - eu-west-1
+          stacks:
+            - test
+          app: my-application-stack
+          contentDirectory: /tmp/cdk.out
+          parameters:
+            templateStagePaths:
+              TEST: test-stack.template.json
+        my-app-ami-test-eu-west-1:
+          type: ami-cloudformation-parameter
```

Comment:
Possibly, I couldn't work it out! `ami-cloudformation-parameter` is used because GuCDK adds an AMI parameter per app.

If we had 2 ASGs:
  1. `new GuEc2App(..., ..., { app: "frontend" } )`
  2.`new GuEc2App(..., ..., { app: "api" } )` 

The CFN JSON will have 2 parameters:
  1. AMIFrontend
  2. AMIApi

`ami-cloudformation-parameter` was the only way I could get Riff-Raff to update both parameters.